src/redux/slices/articlesSlice.ts

```diff
@@ -72,7 +72,9 @@ const articlesSlice = createSlice({
   },
 })
 
-export const getAllArticles = (page, token) => async (dispatch) => {
+export const getAllArticles = (page: number, token: string) => async (
+  dispatch: (arg0: { payload: any; type: string }) => void
+) => {
```

Comment:
Te typy dla wszystkich thunkow mozesz sobie uproscic:
`store.ts` zamieniasz na cos takiego:

```ts
import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'

import rootReducer from "./rootReducer"
import { loadState } from "lib/utils/localStorage"

const preloadedState = loadState("state")

export const store = configureStore({
  reducer: rootReducer,
  preloadedState,
})

export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
  >

export default store

```

Tu jest troche magii reduxowej, ale interestuje nas AppThunk, bo dzieki temu te trzy linijki mozesz zamienic na te:

```ts
export const getArticlesFeeds = (page: number, token: string): AppThunk => async (
  dispatch
) => {
```
Wiec nie musisz typowac sobie pojedynczo argumentow dispatcha, tylko w kazdym miejscu uzywasz `:AppThunk`

Reply:
Na pierwszy rzut oka, to jest straszna magia, faktycznie). Az troche strasznie)